### PR TITLE
fix(security): cap Retry-After sleep and sanitize mimeType in uploads

### PR DESCRIPTION
**Pull request by anshul-garg27**
## Summary

Two security hardening fixes for agent-facing attack surfaces:

### 1. Cap `Retry-After` sleep to 60 seconds (Critical)

A hostile or compromised API server can return `Retry-After: 4294967295` in a 429 response, causing the CLI process to sleep for ~136 billion years. This is a denial-of-service vector, especially dangerous in unattended AI agent workflows where the process runs without human supervision.

**Fix:** Add `.min(60)` cap on the parsed `Retry-After` value in `send_with_retry()`.

### 2. Sanitize `mimeType` in multipart upload headers (High)

The user-supplied `mimeType` field from `--json` body metadata is embedded directly into a MIME `Content-Type` header:
```
Content-Type: {media_mime}\r\n\r\n
```

A crafted `mimeType` like `text/plain\r\nX-Injected: malicious` could inject arbitrary MIME headers into the outbound multipart request body.

**Fix:** Strip `\r` and `\n` characters from `media_mime` before header construction.

## Test plan

- [x] Added test verifying retry cap clamps large values to 60
- [x] Added test verifying normal values pass through unchanged
- [x] Existing client and executor tests continue to pass

### CODE REVIEW

File: src/client.rs [hunk 1]
```diff
@@ -56,4 +60,16 @@ mod tests {
...
     }
+
+    #[test]
+    fn retry_after_cap_prevents_unbounded_sleep() {
+        // Verify the cap constant is reasonable
+        const MAX_RETRY_DELAY_SECS: u64 = 60;
+        // A server sending Retry-After: 999999 should be capped to 60
+        let server_value: u64 = 999_999;
+        let capped = server_value.min(MAX_RETRY_DELAY_SECS);
+        assert_eq!(capped, 60);
```
[the remark this answers is not in the document]
Comment (anshul-garg27) [reply src/client.rs]: Fixed — moved `MAX_RETRY_DELAY_SECS` to module level and replaced the magic number `60` in the test assertion with the constant. Verified locally with `cargo test`.

File: src/client.rs [hunk 2]
```diff
@@ -56,4 +61,14 @@ mod tests {
...
+
+    #[test]
+    fn retry_after_cap_prevents_unbounded_sleep() {
+        // A server sending Retry-After: 999999 should be capped
+        let server_value: u64 = 999_999;
+        let capped = server_value.min(MAX_RETRY_DELAY_SECS);
+        assert_eq!(capped, MAX_RETRY_DELAY_SECS);
+        // Normal values below the cap pass through unchanged
+        assert_eq!(5u64.min(MAX_RETRY_DELAY_SECS), 5);
+    }
```
[the remark this answers is not in the document]
Comment (anshul-garg27) [reply src/client.rs]: Fixed — extracted `compute_retry_delay()` as a separate testable function with 5 focused tests: large value capping, small passthrough, missing header fallback, unparseable header fallback, and boundary value (60 vs 61). All verified locally.

File: src/executor.rs [hunk 3]
```diff
@@ -737,12 +737,17 @@ fn build_multipart_body(
...
+    // Strip CR/LF to prevent MIME header injection via user-controlled mimeType.
+    let media_mime_raw = metadata
         .as_ref()
         .and_then(|m| m.get("mimeType"))
         .and_then(|v| v.as_str())
         .unwrap_or("application/octet-stream");
+    let media_mime: String = media_mime_raw
+        .chars()
+        .filter(|c| *c != '\r' && *c != '\n')
+        .collect();
```
[the remark this answers is not in the document]
Comment (anshul-garg27) [reply src/executor.rs:776]: Fixed — added `test_build_multipart_body_sanitizes_mime_injection` that passes a mimeType with embedded \\r\\n and verifies the CRLF is stripped, preventing header injection. Verified locally.

File: src/client.rs [hunk 4]
```diff
@@ -48,6 +50,16 @@ pub async fn send_with_retry(
     build_request().send().await
 }
 
+/// Compute the retry delay from a Retry-After header value and attempt number.
+/// Falls back to exponential backoff (1, 2, 4s) when the header is absent or
+/// unparseable. Always caps the result at MAX_RETRY_DELAY_SECS.
+fn compute_retry_delay(header_value: Option<&str>, attempt: u32) -> u64 {
+    header_value
+        .and_then(|s| s.parse::<u64>().ok())
+        .unwrap_or(1 << attempt)
```
[the remark this answers is not in the document]
Comment (anshul-garg27) [reply src/client.rs]: Fixed — changed `1 << attempt` to `1u64 << attempt` to prevent potential i32 overflow. Good catch.

File: src/executor.rs [hunk 5]
```diff
@@ -737,12 +737,17 @@ fn build_multipart_body(
...
+    // Determine the media MIME type from the metadata's mimeType field, or fall back.
+    // Strip CR/LF to prevent MIME header injection via user-controlled mimeType.
+    let media_mime_raw = metadata
         .as_ref()
         .and_then(|m| m.get("mimeType"))
         .and_then(|v| v.as_str())
         .unwrap_or("application/octet-stream");
+    let media_mime: String = media_mime_raw
+        .chars()
+        .filter(|c| *c != '\r' && *c != '\n')
```
[the remark this answers is not in the document]
Comment (anshul-garg27) [reply src/executor.rs]: Fixed — now using `.filter(|c| !c.is_control())` to strip all control characters, not just CR/LF. This covers null bytes, tabs, vertical tabs, form feeds, etc.

File: src/client.rs [hunk 6]
```diff
@@ -33,13 +36,12 @@ pub async fn send_with_retry(
...
-        // Parse Retry-After header (seconds), fall back to exponential backoff
-        let retry_after = resp
+        let header_value = resp
             .headers()
             .get("retry-after")
             .and_then(|v| v.to_str().ok())
-            .and_then(|s| s.parse::<u64>().ok())
-            .unwrap_or(1 << attempt); // 1, 2, 4 seconds
+            .map(|s| s.to_string());
+        let retry_after = compute_retry_delay(header_value.as_deref(), attempt);
```
[the remark this answers is not in the document]
Comment (anshul-garg27) [reply src/client.rs]: Fixed — removed the unnecessary `.map(|s| s.to_string())` and `.as_deref()`. Now passes the `&str` directly from `HeaderValue::to_str()`.

File: src/client.rs [hunk 7]
```diff
@@ -48,6 +49,16 @@ pub async fn send_with_retry(
     build_request().send().await
 }
 
+/// Compute the retry delay from a Retry-After header value and attempt number.
+/// Falls back to exponential backoff (1, 2, 4s) when the header is absent or
+/// unparseable. Always caps the result at MAX_RETRY_DELAY_SECS.
+fn compute_retry_delay(header_value: Option<&str>, attempt: u32) -> u64 {
+    header_value
+        .and_then(|s| s.parse::<u64>().ok())
+        .unwrap_or(1u64 << attempt)
```
[the remark this answers is not in the document]
Comment (anshul-garg27) [reply src/client.rs]: Fixed — replaced `1u64 << attempt` with `2u64.saturating_pow(attempt)`. Saturates to `u64::MAX` instead of panicking if attempt >= 64, and gets capped to 60s by `.min(MAX_RETRY_DELAY_SECS)` anyway.

File: src/executor.rs [hunk 8]
```diff
@@ -763,12 +763,17 @@ fn build_multipart_body(
...
+    // Strip CR/LF to prevent MIME header injection via user-controlled mimeType.
+    let media_mime_raw = metadata
         .as_ref()
         .and_then(|m| m.get("mimeType"))
         .and_then(|v| v.as_str())
         .unwrap_or("application/octet-stream");
+    let media_mime: String = media_mime_raw
+        .chars()
+        .filter(|c| !c.is_control())
+        .collect();
```
[the remark this answers is not in the document]
Comment (anshul-garg27) [reply src/executor.rs:776]: good edge case — added a fallback to application/octet-stream if the sanitized mime ends up empty

**Comment by jpoehnelt** [comment]
Superseded by #523 (merged as b241a5b) — same changes shipped.
